### ISSUE TITLE
[BUG] Default collation in postgresql flexible fails

### ISSUE THREAD

**Issue by dnomikos-kar** [claranet/terraform-azurerm-db-postgresql-flexible#3]
### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.4.4

### AzureRM Provider Version

3.73.0

### Affected Resource(s)/Data Source(s)

azurerm_postgresql_flexible_server_database

### Terraform Configuration Files

```hcl
module "db-postgresql-flexible" {
  source  = "claranet/db-postgresql-flexible/azurerm"
  version = "7.0.0"

  administrator_login    = "admin"                                                                                             
  administrator_password = "password"
  allowed_cidrs = {
    "all"   = "0.0.0.0/32"
  }

  custom_server_name    = "postgresql-flexible-server"
  client_name           = ""
  environment           = ""
  location              = "West Europe"
  location_short        = ""
  logs_destinations_ids = []
  resource_group_name   = "resource_group"
  stack                 = ""
  postgresql_version    = 13

  storage_mb   = 131072
  standby_zone = null

  postgresql_configurations = {}
  databases = {
    main = {
      collation = "en_US.UTF8"
      charset   = "UTF8"
    }
  }
}
```

### Debug Output/Panic Output

```shell
│ Error: collation contains unknown collation en_US.UTF8
│ 
│   with module.db-postgresql-flexible[0].azurerm_postgresql_flexible_server_database.postgresql_flexible_db["main"],
│   on .terraform/modules/db-postgresql-flexible/r-postgresql-flexible.tf line 59, in resource "azurerm_postgresql_flexible_server_database" "postgresql_flexible_db":
│   59:   collation = each.value.collation
```

### Expected Behaviour

The resource should be successfully be created

### Actual Behaviour

This should work until azurerm 3.69.0 version. They introduced some validation on the collation field that the default "en_US.UTF8" doesn't comply any more. The value "en_US.utf8" passes the validation and works but forces the complete replacement of the resource which isn't favorable for many production scenarios. It would be nice to have the collation on ignore_changes lifecycle maybe.

More information can be found in this issue in the AzureRM provider repository:
https://github.com/hashicorp/terraform-provider-azurerm/issues/22936

### Steps to Reproduce

1. terraform apply

**Comment by mloskot** [comment]
I also experienced this issue with the latest version 7.1.1 of the module

I've proposed PR with the fix
- #4

**Comment by mloskot** [comment]
For the record, it's now been fixed in [v7.2.0](https://registry.terraform.io/modules/claranet/db-postgresql-flexible/azurerm/7.2.0).
